File: ch7/7-9.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Printf("%t\n", isupper1('G'))
	fmt.Printf("%t\n", isupper1('t'))
	fmt.Printf("%t\n", isupper2('K'))
	fmt.Printf("%t\n", isupper2('m'))
}

func isupper1(c rune) bool {
	return c >= 'A' && c <= 'Z'
}

func isupper2(c rune) bool {
	s := "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
	return strings.ContainsRune(s, c)
}

File: kr/ch7/7-9.c
#include <stdio.h>
#include <string.h>

int isupper1(int c);
int isupper2(int c);

int main(void)
{
    printf("%d\n", isupper1('G'));
    printf("%d\n", isupper1('t'));
    printf("%d\n", isupper2('K'));
    printf("%d\n", isupper2('m'));
    return 0;
}

int isupper1(int c)
{
    return (c >= 'A' && c <= 'Z');
}

int isupper2(int c)
{
    return (strchr("ABCDEFGHIJKLMNOPQRSTUVWXYZ", c) != NULL);
}
